manager/src/main/java/io/gingersnapproject/RuleResource.java

```
@@ -47,4 +59,9 @@ public Multi<String> getAllKeys(String rule) {
    public Uni<QueryResult> query(@QueryParam("query") String query) {
       return queryHandler.query(query);
    }
+
+   private void checkRuleExists(String rule) {
+      if (ruleManager.getEagerOrLazyRuleByName(rule) == null)
+         throw new NotFoundException();
```

Have to admit, I am glad this works, because this seems pretty dirty throwing an exception instead of returning a Multi completed with the exception. Leave it as is though since it is simpler and test  confirms it works still.